compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala

```diff
@@ -1180,6 +1255,78 @@ class JSInteropTest extends DirectTest with TestHelpers {
   }
 
   @Test
+  def noJSSymbolNameOnTopLevelClassesAndObjects: Unit = {
+    for {
+      kind <- Seq("class", "object")
+    } {
+      s"""
+      object Sym {
+        val sym = js.Symbol()
+      }
+
+      @JSName(Sym.sym)
+      @js.native
+      $kind A extends js.Object
+      """ hasErrors
+      """
+        |newSource1.scala:9: error: @JSName with a js.Symbol can only be used on members of JavaScript types
+        |      @JSName(Sym.sym)
+        |       ^
+      """
+    }
+  }
+
+  @Test
+  def noJSSymbolNameOnNestedNativeClassesAndObjects: Unit = {
+    for {
+      kind <- Seq("class", "object")
+    } {
+      s"""
+      object Sym {
+        val sym = js.Symbol()
+      }
+
+      @js.native
+      object Enclosing extends js.Object {
+        @JSName(Sym.sym)
+        @js.native
+        $kind A extends js.Object
+      }
+      """ hasErrors
+      """
+        |newSource1.scala:11: error: Implementation restriction: @JSName with a js.Symbol is not supported on nested native classes and objects
+        |        @JSName(Sym.sym)
+        |         ^
+      """
```

Comment:
The error message above does not work here, because `A` really is a member of a JS type (`Enclosing`).